x-pack/plugins/observability_solution/investigate_app/public/hooks/use_fetch_events.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useQuery } from '@tanstack/react-query';
+import { GetEventsResponse } from '@kbn/investigation-shared';
+import { investigationKeys } from './query_key_factory';
+import { useKibana } from './use_kibana';
+
+export interface Response {
+  isInitialLoading: boolean;
+  isLoading: boolean;
+  isRefetching: boolean;
+  isSuccess: boolean;
+  isError: boolean;
+  data?: GetEventsResponse;
+}
+
+export function useFetchEvents({
+  rangeFrom,
+  rangeTo,
+}: {
+  rangeFrom?: string;
+  rangeTo?: string;
+}): Response {
+  const {
+    core: {
+      http,
+      notifications: { toasts },
+    },
+  } = useKibana();
+
+  const { isInitialLoading, isLoading, isError, isSuccess, isRefetching, data } = useQuery({
+    queryKey: investigationKeys.events(),
+    queryFn: async ({ signal }) => {
+      return await http.get<GetEventsResponse>(`/api/observability/events`, {
+        query: {
+          rangeFrom,
+          rangeTo,
+        },
+        version: '2023-10-31',
+        signal,
+      });
+    },
+    cacheTime: 600 * 1000, // 10_minutes
```

Comment:
I noticed that when changing the timerange on investigation page, the hook doesn't update list of events even if there are new events outside of initial time range. I think we need to find another solution to cache already detected events. For now, maybe we can remove caching from hook, wdyt?

Reply:
That's because the queryKey is not using the rangeFrom and rangeTo parameters. Therefore the cache entry is always the same, and react query simply returns the cached response.